src/main/java/graphql/nadel/Nadel.java

```diff
@@ -117,16 +118,18 @@ private Link createHydrationLink(SchemaNamespace schemaNamespace, FieldDefinitio
         String targetTypeName = TypeInfo.typeInfo(targetField.getType()).getName();
 
         //TODO: braid does not support multiple arguments, so we just take first
-        final String fromField = hydration.getArguments().entrySet().stream()
-                .map(e -> e.getValue().getInputName())
+        final RemoteArgumentDefinition argument = hydration.getArguments().stream()
                 .findFirst()
-                .orElse(definition.field().getName());
+                // braid does not allow links to 0 argument queries. It must be exactly one
+                .orElseThrow(() -> new InvalidDslException("Remote argument is required.",
+                        hydration.getSourceLocation()));
 
-        return Link.from(schemaNamespace, definition.parentType(), definition.field().getName(), fromField)
+        return Link.from(schemaNamespace, definition.parentType(), definition.field().getName(),
+                argument.getFieldMappingDefinition().getInputName())
                 //TODO: we need to add something to DSL to support 'queryVariableArgument' parameter of .to
                 // by default it is targetField name which is not always correct.
                 .to(targetService, targetTypeName, targetField.getName())
-                .replaceFromField()
+                .argument(argument.getName())
```

Comment:
why did you remove 'replaceFromField'?

Reply:
In nadel DSL you define public schema, so it is up to you if you want to include from field or not(you either put it there or not, it is not needed for hydration as long as it is defined in private schema). If replaceFromField will be here there will be no way of including source field AND hydrated field that uses it.

Reply:
👍